dallinger/prolific.py

```diff
@@ -55,41 +55,71 @@ def add_participants_to_study(self, study_id: int, number_to_add: int) -> dict:
+    def approve_participant_submission(self, submission_id: str) -> dict:
         """Mark an assignment as approved.
 
         We do some retrying here, because our first attempt to approve will
         happen more or less simultaneously with the worker submitting
         the study on Prolific. If we get there first, there will be an error
         because the submission hasn't happened yet.
         """
-        status = self.get_participant_session(session_id)["status"]
+        status = self.get_participant_submission(submission_id)["status"]
         if status != "AWAITING REVIEW":
             # This will trigger a retry from the decorator
             raise ProlificServiceException("Prolific session not yet submitted.")
 
         return self._req(
             method="POST",
-            endpoint=f"/submissions/{session_id}/transition/",
+            endpoint=f"/submissions/{submission_id}/transition/",
             json={"action": "APPROVE"},
         )
 
-    def get_participant_session(self, session_id: str) -> dict:
-        """Retrieve details of a participant Session
+    def get_participant_submission(self, submission_id: str) -> dict:
+        """Retrieve details of a participant Submission
+
+        See: https://docs.prolific.com/docs/api-docs/public/#tag/Submissions/Submission-object
 
         This is roughly equivalent to an Assignment on MTurk.
 
         Example return value:
 
         {
-            "id": "60d9aadeb86739de712faee0",
-            "study_id": "60aca280709ee40ec37d4885",
-            "participant": "60bf9310e8dec401be6e9615",
+            "assignment_id": "60d9aadeb86739de712faee0",
+            "hit_id": "60aca280709ee40ec37d4885",
+            "worker_id": "60bf9310e8dec401be6e9615",
             "started_at": "2021-05-20T11:03:00.457Z",
-            "status": "ACTIVE"
+            "status": "ACTIVE",
+        }
+        """
+        response = self._req(method="GET", endpoint=f"/submissions/{submission_id}/")
+        if response:
+            return _translate_submission(response)
+
+    def get_assignments_for_study(self, study_id: str) -> dict:
+        """Return all submissions for the current Prolific study, keyed by
+        assignment (Prolific "submission") ID.
+
+        Example return value:
+
+        {
+          "results": [
+            {
+              "id": "60d9aadeb86739de712faee0",
+              "participant_id": "60bf9310e8dec401be6e9615",
+              "started_at": "2021-05-20T11:03:00.457000Z",
+              "status": "ACTIVE",
+              "study_code": "ABC123"
+            }
+          ]
         }
         """
-        return self._req(method="GET", endpoint=f"/submissions/{session_id}/")
+
+        query_params = {"study": study_id}
+        response = self._req(
+            method="GET", endpoint="/submissions/", params=query_params
+        )
+
+        return {s["participant"]: _translate_submission(s) for s in response["results"]}
```

Comment:
The structure of a submission passed to `_translate_submission` included in the responses of these two API calls looks different. So we can't use the same function to map a submission to Dallinger terminology, See 
https://docs.prolific.com/docs/api-docs/public/?_gl=1*uk7kh0*_gcl_au*MTAwMDM1NTU2MS4xNzI2MDY3MTcy#tag/Submissions/operation/GetSubmission
and
https://docs.prolific.com/docs/api-docs/public/?_gl=1*uk7kh0*_gcl_au*MTAwMDM1NTU2MS4xNzI2MDY3MTcy#tag/Submissions/operation/GetSubmissions

Reply:
Also, what does `study_code` refer to in the API response. It looks different than a `study_id`.

Reply:
Resolved.